### ISSUE TITLE
documentation should mention issues of fork and multi-threading

### ISSUE THREAD

**Issue by MartinNowak** [NCrashed/daemonize#2]
http://forum.dlang.org/post/ksqubftqniwznqbmurrk@forum.dlang.org
http://www.linuxprogrammingblog.com/threads-and-fork-think-twice-before-using-them

**Comment by NCrashed** [comment]
Hmm, that a complex problem. Could it be avoided by using an `exec` function after forking?

**Comment by MartinNowak** [comment]
Yes, I'd think so. If you can restart the daemonized process using exec, then it should avoid the problem.

**Comment by etcimon** [comment]
GC deadlock issue on the forums was resolved here: https://github.com/etcimon/libasync/pull/22

**Comment by MartinNowak** [comment]
It should still be mentioned in the readme and/or the wiki

**Comment by andre2007** [comment]
The information here are a little bit confusing for me:
- Is this still an issue on linux? 
- Does the issue occur only when I do multi threading in general or only when using libasync?

MartinNowak mentioned the issue simply can be solved by using an exec function after forking. I am not an expert here, but if this is just a few lines, could you  add these line if you have some time?
